### PR TITLE
Fix get-password for gce

### PR DESCRIPTION
**Pull request by j3ffml**
#4094 Tested by deleting local ~/.kubernetes_auth and spinning up an e2e cluster. I had two failures running e2e, one of which may be a flake (pd.sh), but this should at least fix the cluster validation problem for jenkins.

cc @zmerlynn, @brendandburns

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -189,13 +189,26 @@ function detect-master () {
...
+import yaml, sys
+cfg = yaml.load(sys.stdin)
+try:
+  current = cfg["current-context"]
+  user = cfg["contexts"][current]["user"]
+  print cfg["users"][user]["auth-path"]
+except KeyError:
+  pass
+'''
+  local file=$("${KUBE_ROOT}/cluster/kubectl.sh" config view --global | python -c "${get_auth_path}")
```
Comment (zmerlynn) [inline cluster/gce/util.sh]: I would love to express my displeasure at inline python for this solution, but it's the most readable inline python in our codebase so far.

Something something need a YAML splitter for bash blah blah rant rant except this should be in go anyways LGTM.

**Comment by zmerlynn** [comment]
Merging after hours to hope for some e2e runs.

**Comment by brendandburns** [comment]
This appears not to have fixed things.  Do we need to worry about file locations in the jenkins install?

**Comment by brendandburns** [comment]
Or perhaps there is more than one problem...  I'm looking at this.

**Pull request merged**